packages/core/src/validate.js

```diff
@@ -80,21 +80,22 @@ function toErrorSchema(errors) {
   }, {});
 }
 
-export function toErrorList(errorSchema, fieldName = "root") {
-  // XXX: We should transform fieldName as a full field path string.
+export function toErrorList(errorSchema, fieldPath = []) {
```

Comment:
Sorry, I'm a bit confused as to what the changes in `toErrorList` have to do with fixing #1596. Doesn't the change in line 259 already ensure that the original error list is untouched?

Reply:
This is for the comment above:
```
Note how the stack message is subtly different here:
RJSF custom errors use fieldName: message
AJV errors use .field.path message
```

toErrorList only tracked the current field's name, not the full path to that field.  I changed the function to receive the full field path so that it can use the dot-separated field path when generating the message.

Reply:
Ah, thanks for the pointer.